podman_hpc/siteconfig.py

```diff
@@ -2,55 +2,193 @@
 import os
 import shutil
 import toml
-import yaml
+import re
+from yaml import load
+from yaml import FullLoader
 from copy import deepcopy
 from glob import glob
 
-_MOD_ENV = "PODMANHPC_MODULES_DIR"
-_HOOKS_ENV = "PODMANHPC_HOOKS_DIR"
+_ENV_PREFIX = "PODMANHPC"
+_MOD_ENV = f"{_ENV_PREFIX}_MODULES_DIR"
 _HOOKS_ANNO = "podman_hpc.hook_tool"
+_CONF_ENV = f"{_ENV_PREFIX}_CONFIG_FILE"
 
 
 class SiteConfig:
     """
     Class to represent site specific configurations for Podman-HPC.
+
+    Precedence order lowest to highest:
+    - Built-in defaults
+    - Config file
+    - Environment variables
     """
 
+    _default_conf_file = "/etc/podman_hpc/podman_hpc.yaml"
```

Comment:
Should podman-hpc generate a default podman_hpc.yaml? Or is this something we expect sites to put into place (in our case via ansible)?

Reply:
Ah reading this again I see the defaults take precedence, so the config file is optional. Never mind, this looks good.